src/scp/test/SCPTests.cpp

```diff
@@ -2821,6 +2821,48 @@ TEST_CASE("nomination tests core5", "[scp][nominationprotocol]")
                 verifyNominate(scp.mEnvs[1], v0SecretKey, qSetHash0, 0, votesXK,
                                {});
             }
+            SECTION("select accepted value from leader")
+            {
+                REQUIRE(xValue < yValue);
+                REQUIRE(yValue < zValue);
+                REQUIRE(scp.mEnvs.size() == 1);
+
+                // Update round leader to v1
+                scp.mPriorityLookup = [&](NodeID const& n) {
```

Comment:
You probably want this section to be under `SECTION("v1 is top node")` so that you don't end up with `leaders {= v0, v1}`

Reply:
actually we probably want to check both cases (v0 leader first and v1 leader from the beginning)?

Reply:
not sure I follow why both are needed. In the current check, we know exactly who the leaders are and what they voted for. And since the values are all different, we just need to make sure the updated function is called correctly and the right values are selected.

Reply:
yes you are correct